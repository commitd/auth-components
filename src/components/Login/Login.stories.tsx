import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Login, LoginProps } from '.'
import { ThemeProvider } from '@committed/components'

export default {
  title: 'Components/Login',
  component: Login,
} as Meta

export const Default: React.FC = () => {
  return (
    <ThemeProvider>
      <Login
        onChangeEmail={() => {}}
        onChangePassword={() => {}}
        email="jsmith"
        password="password"
        onSignIn={() => {}}
      />
    </ThemeProvider>
  )
}

const Template: Story<LoginProps> = (args) => (
  <ThemeProvider>
    <Login {...args} />
  </ThemeProvider>
)

export const Primary = Template.bind({})
Primary.args = {}
