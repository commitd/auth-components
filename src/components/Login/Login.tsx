import {
  Box,
  Button,
  Container,
  Form,
  Flex,
  IconButton,
  InputAdornment,
  Text,
  Typography,
  Alert,
} from '@committed/components'
import { TextField } from '@material-ui/core'
import { Visibility } from '@material-ui/icons'
import { VisibilityOff } from '@material-ui/icons'
import React, { useState } from 'react'

export interface LoginProps {
  onChangeEmail: (e: React.ChangeEvent<HTMLInputElement>) => void
  onChangePassword: (e: React.ChangeEvent<HTMLInputElement>) => void
  email: string
  password: string
  errorMessage?: string
  onSignIn: () => void
  passwordRules?: string
  submitting?: boolean
}

export const Login: React.FC<LoginProps> = ({
  errorMessage,
  onChangeEmail,
  onChangePassword,
  email,
  password,
  onSignIn,
  passwordRules,
  submitting,
  children,
}) => {
  const [hidePassword, setHidePassword] = useState(true)
  return (
    <Container component="main" maxWidth="sm">
      <Flex alignItems="center" flexDirection="column">
        {children}
        {errorMessage != null && errorMessage.length > 0 ? (
          <Box mb={3}>
            <Alert severity="error">
              <Text>{errorMessage}</Text>
            </Alert>
          </Box>
        ) : undefined}
        <Form
          width={1}
          onSubmit={(e): void => {
            e.preventDefault()
            onSignIn()
          }}
        >
          <Box mb={3}>
            <TextField
              fullWidth
              required
              id="email"
              label="Email Address"
              name="email"
              autoComplete="email"
              autoFocus
              type="email"
              value={email}
              onChange={onChangeEmail}
              variant="outlined"
            />
          </Box>

          <Box mb={1}>
            <TextField
              fullWidth
              required
              name="password"
              label="Password"
              type={hidePassword ? 'password' : 'text'}
              id="password"
              autoComplete="current-password"
              value={password}
              onChange={onChangePassword}
              aria-describedby="password-contraints"
              variant="outlined"
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <IconButton
                      aria-label="toggle password visibility"
                      onClick={(): void => setHidePassword((v) => !v)}
                      edge="end"
                    >
                      {hidePassword ? <Visibility /> : <VisibilityOff />}
                    </IconButton>
                  </InputAdornment>
                ),
              }}
            />
          </Box>

          {passwordRules != null && passwordRules.length > 0 ? (
            <Box id="password-contraints">
              <Typography variant="caption"> {passwordRules}</Typography>
            </Box>
          ) : undefined}
          <Button
            mt={3}
            color="primary"
            fullWidth
            type="submit"
            disabled={submitting}
          >
            Sign In
          </Button>
        </Form>
      </Flex>
    </Container>
  )
}
