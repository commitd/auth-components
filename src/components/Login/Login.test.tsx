import React from 'react'
import { render } from '../../setupTests'
import { Default } from './Login.stories'

it('renders story', () => {
  const { asFragment } = render(<Default />)
  expect(asFragment()).toMatchSnapshot()
})
