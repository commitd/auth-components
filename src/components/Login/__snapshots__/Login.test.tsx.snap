// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders story 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiBox-root MuiBox-root-3 Styled(MuiBox)-root-1 Styled(MuiBox)-root-2"
    >
      <form
        class="Component-root-6 Component-root-7 Component-root-4 Component-root-5"
      >
        <div
          class="MuiBox-root MuiBox-root-9 Styled(MuiBox)-root-1 Styled(MuiBox)-root-8"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
              data-shrink="true"
              for="email"
              id="email-label"
            >
              Email Address
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="email"
                value="jsmith"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13"
                >
                  <span>
                    Email Address *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-15 Styled(MuiBox)-root-1 Styled(MuiBox)-root-14"
        >
          <div
            aria-describedby="password-contraints"
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
              data-shrink="true"
              for="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="password"
                name="password"
                required=""
                type="password"
                value="password"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root WithNoOpener-root-16 WithNoOpener-root-17 MuiIconButton-edgeEnd"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-12 PrivateNotchedOutline-legendNotched-13"
                >
                  <span>
                    Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root-18 Component-root-19 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Sign In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</DocumentFragment>
`;
